### ISSUE TITLE
restarting app does not use cached credentials

### ISSUE THREAD

**Issue by Paul-Todd** [davecoates/rn-firebase-bridge#24]
I am investigating this further but I noticed on iOS that if I:

1. signin
2. exit the app
3. start the app again

It should skip the signin phase and go straight to the main screen since the current user is retrieved as they are logged in.

In the previous release this worked fine - I called Auth.getCurrentUser() which would return null or the user that is logged in (via a promise).

In the current release this no longer works - I attached an onAuthStateChanged which returns either undefined if the runtime is not initialized or null. Once I called signin then it is correctly populated with the new user.

**Comment by davecoates** [comment]
Ah - on the initial app load this will work fine but once you refresh the app in development mode it will lose the data (everything get's re-instantiated and the auth stage change won't be fired). Previously calling `getCurrentUser()` would work because it actually queried the native SDK for the current user.

Sorry about that, will try to address later today.

**Comment by davecoates** [comment]
Hmm actually it should be OK as the listeners get added and fire again... but you are correct the user is null. Not sure what's going on... will investigate further

**Comment by davecoates** [comment]
Ok so the problem is in `FirebaseBridgeApp.swift` - in `invalidate()` it _deletes_ all apps which causes the loss of authentication details etc. This made it easy to refresh the app because I could safely initialize any app again and be guaranteed that it wasn't already initialized - I hadn't thought through the consequences of deleting each app though! I'll fix this properly soon and make sure it's fine on Android but for now you could probably replace `FirebaseBridgeApp.swift` with this:

```
//  FirebaseApp.swift
//
//  Created by Dave Coates on 25/09/2016.
//

import Firebase

func appToDict(app:FIRApp) -> Dictionary<String, AnyObject> {
  let options = app.options
  let optionsDict = options.dictionaryWithValuesForKeys([
    "androidClientID",
    "APIKey",
    "databaseURL",
    "GCMSenderID",
    "googleAppID",
    "storageBucket",
    ])
  let data:Dictionary<String, AnyObject> = [
    "name": app.name,
    "options" : optionsDict,
  ]
  
  return data
}

@objc(FirebaseBridgeApp)
class FirebaseBridgeApp: NSObject, RCTInvalidating {
  
  func invalidate() {
    print("invalidate")
    /*
    if let apps = FIRApp.allApps() {
      for app in apps {
        (app.1 as! FIRApp).deleteApp({ (success) in
          if (!success) {
            print("Failed to delete app", app)
          }
        })
      }
    }
    */
  }
  
  @objc func initializeApp(options:NSObject, name:String?, resolver resolve: RCTPromiseResolveBlock, rejecter reject: RCTPromiseRejectBlock) {
    
    if let app = FIRApp(named: name ?? "__FIRAPP_DEFAULT") {
      resolve(appToDict(app));
      return
    }
    
    let fbOptions = FIROptions.init(
      googleAppID: options.valueForKey("googleAppID") as? String,
      bundleID: options.valueForKey("bundleID") as? String,
      GCMSenderID: options.valueForKey("GCMSenderID") as? String,
      APIKey: options.valueForKey("APIKey") as? String,
      clientID: options.valueForKey("clientID") as? String,
      trackingID: options.valueForKey("trackingID") as? String,
      androidClientID: options.valueForKey("androidClientID") as? String,
      databaseURL: options.valueForKey("databaseURL") as? String,
      storageBucket: options.valueForKey("storageBucket") as? String,
      deepLinkURLScheme: options.valueForKey("deepLinkURLScheme") as? String
    )
    var app:FIRApp?;
    if let name = name where name != "" {
      FIRApp.configureWithName(name, options: fbOptions)
      app = FIRApp(named: name)
    } else {
      //FIRApp.configure()
      FIRApp.configureWithOptions(fbOptions)
      app = FIRApp.defaultApp()
    }
    if let app = app {
      resolve(appToDict(app));
    } else {
      reject("app_initialize_failure", "App initialization failed", NSError(domain: "FirebaseBridge", code: 0, userInfo: nil))
    }
  }
  
  
  @objc func initializeDefaultApp(resolve: RCTPromiseResolveBlock, rejecter reject: RCTPromiseRejectBlock)
  {
    if let app = FIRApp(named:"__FIRAPP_DEFAULT") {
      resolve(appToDict(app));
      return
    }
    
    FIRApp.configure();
    if let app = FIRApp.defaultApp() {
      print(app.name)
      resolve(appToDict(app));
    } else {
      reject("app_initialize_failure", "App initialization failed", NSError(domain: "FirebaseBridge", code: 0, userInfo: nil))
    }
  }

}
```

**Comment by davecoates** [comment]
I've pushed a fix - are you able to try it before I publish? (also pushed fixed to #25)

**Comment by davecoates** [comment]
Available in 0.1.1
